File: Ng-Test/Controllers/CourseController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Ng_Test.Models;

namespace Ng_Test.Controllers
{
    public class CourseController : Controller
    {
        StudentEntities db = new StudentEntities();
        // GET: Course
        public ActionResult Index()
        {
            return View();
        }

        //Get course result
        public JsonResult GetCourse()
        {
            List<Course> courses = db.Courses.ToList();
            List<CourseDTO> sdio = courses.Select(x => new CourseDTO
            {
                CourseId = x.CourseId,
                CourseName = x.CourseName,
                CourseCredit = x.CourseCredit

            }).ToList();
            return Json(sdio, JsonRequestBehavior.AllowGet);
        }

        // Insert Course data 
        public JsonResult InserCourse(Course course)
        {
            if (course != null)
            {
                db.Courses.Add(course);
                db.SaveChanges();
                return Json(new { success=true});
            }
            else
            {
                return Json(new{ success=false});
            }
        }

        // public Course update

        public JsonResult CourseUpdate(Course course)
        {
            var i = db.Courses.Find(course.CourseId);
            if (i == null)
            {
                return Json(new { success=false});
            }
            else
            {
                i.CourseName = course.CourseName;
                i.CourseCredit = course.CourseCredit;
                db.SaveChanges();
                return Json(new { success=true});
            }
        }
    }
}

File: Ng-Test/AngularJsApp/coursecontroller.js
myapp.controller('course_controller', function ($scope, student_service) {
    loadcourse();
    function loadcourse() {
        var courserecord = student_service.getcourse();

        courserecord.then(function (d) {

            $scope.course = d.data;
        },
            function () {
                alert("error occured");
            }
        );
    }

    $scope.courseInsert = function () {
        var coursedata = {
            CourseName: $scope.CourseName,
            CourseCredit: $scope.CourseCredit,
        };
        var saverecord = student_service.Insercourse(coursedata);
        saverecord.then(function (d) {
            if (d.data.success = true) {
                loadcourse();
                alert("Data Insert Successfully....");
                $scope.clear();

            }
            else {
                alert("Data not insert");
            }
        }, function () {
                alert("Error occured while adding Course.");

        });

    }

    $scope.updatecourse = function () {
        var course = {
            CourseId: $scope.CourseId,
            CourseName: $scope.CourseName,
            CourseCredit: $scope.CourseCredit,
        };
        var updaterecord = student_service.courseupdate(course);
        updaterecord.then(function (d) {
            if (d.data.success = true) {
                loadcourse();
                alert("Update Course Info successfully ...");
                $scope.clear();
            }
            else
            {
                alert("Data not update ...");
            }
        }, function () {

                alert("Error occured while update course information...");
        })
    }
    $scope.clear = function () {
        $scope.CourseId = "";
        $scope.CourseName = "";
        $scope.CourseCredit = "";

    }
});